libvast/src/system/type_registry.cpp

```diff
@@ -64,13 +65,24 @@ caf::error type_registry_state::load_from_disk() {
   return caf::none;
 }
 
-void type_registry_state::insert(vast::record_type layout) {
+void type_registry_state::insert(vast::type layout) {
   [[maybe_unused]] auto [hint, success]
     = data[layout.name()].insert(std::move(layout));
   if (success)
     VAST_VERBOSE(self, "registered", hint->name());
 }
 
+std::unordered_set<vast::type> type_registry_state::lookup() const {
+  auto result = std::unordered_set<vast::type>{};
+  // TODO: Replace merging logic once libc++ implements unordered_set::merge.
+  // for ([[maybe_unused]] auto& [key, value] : data)
+  //   result.merge(value);
+  for ([[maybe_unused]] auto& [k, v] : data)
```

Comment:
There's also `detail/set_operations.hpp` with some potentially useful set fu.

Reply:
Sadly not for this use case. It's not too important anyways, the resulting set isn't all too large.

Reply:
The pseudo-code above is also wrong, it should be just `result.merge(data)`